### DISCUSSION TITLE
Flutter CI/CD with Integration Test

### DISCUSSION THREAD

**Discussion by GaGuarGo** [codemagic-ci-cd/codemagic-docs#2766] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

i ran the command flutter test integration_test --flavor prod on code magic. My integration tests are in integration_test folder, and i also have a test_driver folder with a integration_driver.dart as the code magic projetc sample.

### Expected results

to identify an run the integration tests.

### Actual results

Running pod install...                                             85.3s
Running Xcode build...                                          
Xcode build done.                                           108.0s
Failed to build iOS app
Error (Xcode): Framework 'Pods_Runner' not found

Error (Xcode): Linker command failed with exit code 1 (use -v to see invocation)

Could not build the application for the simulator.
{"testID":1,"error":"TestDeviceException(Unable to start the app on the device.)","stackTrace":"package:flutter_tools/src/test/integration_test_device.dart 63:7  IntegrationTestTestDevice.start\n","isFailure":false,"type":"error","time":206962}
{"testID":1,"error":"TimeoutException after 0:12:00.000000: Test timed out after 12 minutes.","stackTrace":"package:test_api/src/backend/invoker.dart 338:28  Invoker._handleError.<fn>\ndart:async/zone.dart 1391:47                      _rootRun\ndart:async/zone.dart 1301:19                      _CustomZone.run\npackage:test_api/src/backend/invoker.dart 336:10  Invoker._handleError\npackage:test_api/src/backend/invoker.dart 291:9   Invoker.heartbeat.<fn>.<fn>\ndart:async/zone.dart 1399:13                      _rootRun\ndart:async/zone.dart 1301:19                      _CustomZone.run\npackage:test_api/src/backend/invoker.dart 290:38  Invoker.heartbeat.<fn>\ndart:async-patch/timer_patch.dart 18:15           Timer._createTimer.<fn>\ndart:isolate-patch/timer_impl.dart 398:19         _Timer._runTimers\ndart:isolate-patch/timer_impl.dart 429:5          _Timer._handleMessage\ndart:isolate-patch/isolate_patch.dart 184:12      _RawReceivePort._handleMessage\n","isFailure":false,"type":"error","time":720039}
{"testID":1,"result":"error","skipped":false,"hidden":false,"type":"testDone","time":720042}
{"success":false,"type":"done","time":720045}

And doesn't even recognize and run the integration tests. The command i'm using is correct? or do i need to use another one?

**Comment by dam-ease** [comment, marked as answer]
Hello @GaGuarGo, 
This error seems to be related to the flutter framework as per this [GH issue](https://github.com/flutter/flutter/issues/105913), and the workaround is to use flutter drive command. Can you please try this out?
If you run into further issues, kindly share the build ID with me. You can find it under your build details page's 'ID' section. ﻿Navigate to the 'Builds' section on the left-hand menu, select the build you want to view, and click the copy icon.
